src/submodule.c

```diff
@@ -1566,6 +1617,33 @@ int git_submodule_location(unsigned int *location, git_submodule *sm)
 		location, NULL, NULL, NULL, sm, GIT_SUBMODULE_IGNORE_ALL);
 }
 
+int git_repository_submodule_cache_all(git_repository *repo)
```

Comment:
Please don't put `git_repository` functions in a file that is not `repository.c`.  Either:  1) these should either go in `repository.c`, or 2) they should be `git_submodule_cache_all` and `git_submodule_cache_clear`.  The latter is obviously more brief, but I think (1) makes more sense.